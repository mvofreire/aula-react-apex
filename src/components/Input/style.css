.input{
  padding:10px;
  margin-right: 5px;
  width: 70%;
}