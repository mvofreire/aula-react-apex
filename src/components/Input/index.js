import React, { forwardRef } from 'react'
import "./style.css"

function Input({ onInputChange, onEnter, ...rest }, ref) {

  const handleOnKeyUp = (e) => {
    const { key } = e

    if (key === 'Enter') {
      onEnter(e);
    } else {
      return true;
    }
  }

  return (
    <input ref={ref} className='input' onChange={onInputChange} onKeyDown={handleOnKeyUp} {...rest} />
  )
}

export default forwardRef(Input)