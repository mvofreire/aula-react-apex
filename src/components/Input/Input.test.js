import { render, act, fireEvent, cleanup } from '@testing-library/react';
import Input from '.';

afterEach(cleanup)

describe('Input', () => {
  test('Should render in screen', () => {
    const { container } = render(<Input />)
    expect(container).toBeInTheDocument()
  })

  test('Should match with snapshot', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot('default')
  })

  test('Should call callback onChange', () => {
    const callback = jest.fn();
    const { container } = render(<Input onInputChange={callback} />)

    act(() => {
      fireEvent.change(container.querySelector('.input'), { target: { value: 'teste' } })
    })

    expect(callback).toBeCalledTimes(1)
  })

  test('Should call callback onEnter', () => {
    const callback = jest.fn();
    const { container } = render(<Input onEnter={callback} />)

    act(() => {
      fireEvent.keyDown(container.querySelector('.input'), { key: 'Enter' })
    })

    expect(callback).toBeCalledTimes(1)
  })

  test('Should NOT call callback onEnter', () => {
    const callback = jest.fn();
    const { container } = render(<Input onEnter={callback} />)

    act(() => {
      fireEvent.keyDown(container.querySelector('.input'), { key: 'Esc' })
    })

    expect(callback).toBeCalledTimes(0)
  })

})