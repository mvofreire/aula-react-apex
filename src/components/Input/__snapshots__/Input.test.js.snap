// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match with snapshot: default 1`] = `
<div>
  <input
    class="input"
  />
</div>
`;
